connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/a2a/client/impl/A2aMessageSenderImpl.java

```diff
@@ -0,0 +1,111 @@
+import io.camunda.connector.agenticai.a2a.client.model.result.A2aSendMessageResult;
+import io.camunda.connector.agenticai.a2a.client.model.result.A2aTask;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BiConsumer;
+import org.apache.commons.collections4.CollectionUtils;
+
+public class A2aMessageSenderImpl implements A2aMessageSender {
+
+  private static final Duration DEFAULT_POLL_INTERVAL = Duration.ofMillis(500);
+
+  private final DocumentToPartConverter documentToPartConverter;
+  private final A2aSendMessageResponseHandler sendMessageResponseHandler;
+  private final TaskPoller taskPoller;
+  private final A2aSdkClientFactory clientFactory;
+
+  public A2aMessageSenderImpl(
+      DocumentToPartConverter documentToPartConverter,
+      A2aSendMessageResponseHandler sendMessageResponseHandler,
+      TaskPoller taskPoller,
+      A2aSdkClientFactory clientFactory) {
+    this.documentToPartConverter = documentToPartConverter;
+    this.sendMessageResponseHandler = sendMessageResponseHandler;
+    this.taskPoller = taskPoller;
+    this.clientFactory = clientFactory;
+  }
+
+  @Override
+  public A2aSendMessageResult sendMessage(
+      SendMessageOperationConfiguration sendMessageOperation, AgentCard agentCard) {
+    Message message = createMessage(sendMessageOperation);
+    CompletableFuture<A2aSendMessageResult> response = new CompletableFuture<>();
+    BiConsumer<ClientEvent, AgentCard> consumer =
+        (event, ignore) -> {
+          try {
+            A2aSendMessageResult result = sendMessageResponseHandler.handleClientEvent(event);
+            response.complete(result);
+          } catch (Exception e) {
+            response.completeExceptionally(e);
+          }
+        };
+
+    Client client = clientFactory.buildClient(agentCard, consumer);
+    try {
+      client.sendMessage(message);
+    } catch (A2AClientException e) {
+      throw new RuntimeException(e);
+    }
+
+    try {
+      Instant startTime = Instant.now();
+      A2aSendMessageResult result =
+          response.get(sendMessageOperation.timeout().toMillis(), TimeUnit.MILLISECONDS);
+      if (result instanceof A2aSendMessageResult.A2aTaskResult(A2aTask task)
+          && task.status().state().isSubmittedOrWorking()) {
+        Duration timeSpent = Duration.between(startTime, Instant.now());
+        Duration updatedTimeout = sendMessageOperation.timeout().minus(timeSpent);
+        return taskPoller
+            .poll(task.taskId(), client, DEFAULT_POLL_INTERVAL, updatedTimeout)
+            .get(updatedTimeout.toMillis(), TimeUnit.MILLISECONDS);
+      }
+      return result;
+    } catch (TimeoutException e) {
+      // TODO: should be a ConnectorException with a specific error code?
```

Comment:
As mentioned above we should probably do something like that, but could also be a dedicated follow-up dealing with error handling.